syntax = "proto3";
package process_definition_version;





/*
GetProcessDefinitionVersion请求
*/
message GetProcessDefinitionVersionRequest {
    /*
    流程定义实例ID
    */
    string definitionId = 1 ;
    /*
    流程定义版本实例ID(lastest表示最新版本)
    */
    string versionId = 2 ;
}

/*
GetProcessDefinitionVersion返回
*/
message GetProcessDefinitionVersionResponse {
message TaskInfo {
    /*
    任务定义ID(bpmn中定义的任务Id)
    */
    string userTaskId = 1 ;
    /*
    名称
    */
    string taskName = 2 ;
    /*
    是否已设置/绑定表单版本(true已设置,false未设置)
    */
    bool isSet = 3 ;
    /*
    表单Id
    */
    string formId = 4 ;
    /*
    表单名称
    */
    string formName = 5 ;
    /*
    表单版本Id
    */
    string formVersionId = 6 ;
    /*
    表单版本名称
    */
    string formVersionName = 7 ;
    /*
    标准工时
    */
    string taskWorkingTime = 8 ;
    /*
    审批时限(workingDay工作日,calendarDay日历日)
    */
    string approvalLimit = 9 ;
}
        
    
    /*
    流程版本实例Id
    */
    string vInstanceId = 1 ;
    /*
    流程版本作者
    */
    string vCreator = 2 ;
    /*
    流程版本说明
    */
    string vMemo = 3 ;
    /*
    用户任务信息
    */
    repeated TaskInfo taskInfo = 4 ;
    /*
    流程定义实例id
    */
    string instanceId = 5 ;
    /*
    流程定义名称
    */
    string name = 6 ;
    /*
    流程类别
    */
    string category = 7 ;
    /*
    流程作者
    */
    string creator = 8 ;
    /*
    流程说明
    */
    string memo = 9 ;
    /*
    创建时间
    */
    string ctime = 10 ;
    /*
    流程部署Id
    */
    string deploymentId = 11 ;
    /*
    版本名称
    */
    string versionName = 12 ;
    /*
    流程定义id
    */
    string flowableDefinitionId = 13 ;
    /*
    流程定义key
    */
    string flowableDefinitionKey = 14 ;
    /*
    是否为主版本(true 为主版本,false为非主版本)
    */
    bool isMain = 15 ;
    /*
    流程状态(unfinished 未完成, done已完成)
    */
    string state = 16 ;
    /*
    流程部署时间
    */
    string deploymentTime = 17 ;
    /*
    父部署Id
    */
    string parentDeploymentId = 18 ;
    /*
    流程BPMN信息(流程详情需要返回,除此之外可以不返回或返回空字符串)
    */
    string bpmnXML = 19 ;
}

/*
GetProcessDefinitionVersionApi返回
*/
message GetProcessDefinitionVersionResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    process_definition_version.GetProcessDefinitionVersionResponse data = 4 ;
}

